dispatchers.go

```diff
@@ -313,6 +352,90 @@ func workdir(b *Builder, args []string, attributes map[string]bool, flagArgs []s
 	return nil
 }
 
+func parseCommandForHeredoc(originalCommand string, name string) (string, string, error) {
+	// Expected format `RUN <str> <<`
+	firstline := originalCommand
+	command := ""
+	pipeCommand := ""
+	if strings.HasPrefix(firstline, "RUN ") || strings.HasPrefix(firstline, "run ") {
+		if strings.HasPrefix(firstline, "RUN ") {
+			firstline = strings.TrimPrefix(firstline, "RUN ")
+		} else if strings.HasPrefix(firstline, "run ") {
+			firstline = strings.TrimPrefix(firstline, "run ")
+		}
+		seperator := "<<"
+		if strings.Contains(firstline, "<<-"+name) {
+			seperator = "<<-"
+		}
+		if strings.Contains(firstline, seperator+name) {
+			arr := strings.Split(firstline, seperator+name)
+			command = strings.TrimSpace(arr[0])
+			commandSplit := strings.Split(command, " ")
+			if len(commandSplit) > 0 {
+				lastToken := commandSplit[len(commandSplit)-1]
+				if lastToken == "&&" || lastToken == "||" {
+					command = lastToken + " " + command
+				}
+				if len(commandSplit) > 1 {
+					lasttolastToken := commandSplit[len(commandSplit)-2]
+					command = lastToken
+					if lasttolastToken == "&&" || lasttolastToken == "||" {
+						command = lasttolastToken + " " + command
+					}
+				}
+			}
+		}
+		// check if any piped instruction
+		tokens := strings.Split(originalCommand, " ")
```

Comment:
Does this handle quoted whitespace in the passed-in line?  Can we add a test for that?

Reply:
Added a test case in `TestParseHeredocCommand`

Reply:
This code is not needed.